src/movedescriptor.js

```diff
@@ -104,6 +104,20 @@ MoveDescriptor.prototype.toString = function() {
 	return result;
 };
 
+MoveDescriptor.prototype.toUCIString = function() {
+	var result = bt.squareToString(this._from);
+	if(this.isCastling()) {
+		// UCI castling is KxR of same color
```

Comment:
The UCI spec does not mention that. Instead, white short castling is defined as `e1g1`.

See http://wbec-ridderkerk.nl/html/UCIProtocol.html

By the way, this may lead to an ambiguity in some chess960 initial position. I don't know whether UCI engines manage chess960 or not...

Reply:
I think I got this from the lichess implementation of castling in their chessground javascript, it seems to be the convention since chess960 is ambigious when king moves less than two squares in castling. e1g1 is preferred if standard, but kxr works for both standard and 960.

Reply:
https://github.com/official-stockfish/Stockfish/blob/master/src/uci.cpp#L335-L371 (uses https://github.com/official-stockfish/Stockfish/blob/master/src/movegen.cpp#L250) - basically Stockfish is using KxR (e1h1 for standard position) for chess960, and I think that is the convention for chess 960.